bundle/src/main/java/com/adobe/acs/commons/packaging/impl/AssetPackagerServletImpl.java

```diff
@@ -0,0 +1,281 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package com.adobe.acs.commons.packaging.impl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import javax.jcr.RepositoryException;
+
+import com.adobe.acs.commons.packaging.PackageHelper;
+import com.day.cq.dam.api.DamConstants;
+import com.day.cq.dam.commons.util.DamUtil;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.sling.SlingServlet;
+import org.apache.jackrabbit.vault.fs.api.PathFilterSet;
+import org.apache.jackrabbit.vault.fs.filter.DefaultPathFilter;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+
+/**
+ * ACS AEM Commons - Asset Packager Servlet
+ * Servlet end-point used to create packages of pages and the referenced assets.
+ */
+@SuppressWarnings("serial")
+@SlingServlet(
+        methods = {"POST"},
+        resourceTypes = {"acs-commons/components/utilities/packager/asset-packager"},
+        selectors = {"package"},
+        extensions = {"json"}
+)
+public class AssetPackagerServletImpl extends AbstractPackagerServlet {
+
+    /* Property names */
+    private static final String PN_PAGE_PATH = "pagePath";
+    private static final String PN_EXCLUDE_PAGES = "excludePages";
+    private static final String PN_ASSET_PREFIX = "assetPrefix";
+    private static final String PN_PAGE_EXCLUSIONS = "pageExclusions";
+    private static final String PN_ASSET_EXCLUSIONS = "assetExclusions";
+
+    /* Default Properties */
+    private static final String DEFAULT_PACKAGE_NAME = "assets";
+    private static final String DEFAULT_PACKAGE_GROUP_NAME = "Assets";
+    private static final String DEFAULT_PACKAGE_DESCRIPTION = "Asset Package initially defined by a ACS AEM Commons - "
+            + "Asset Packager configuration.";
+    private static final String THUMBNAIL_RESOURCE_PATH =
+            "/apps/acs-commons/components/utilities/packager/asset-packager/definition/package-thumbnail.png";
+
+    @Reference
+    private PackageHelper packageHelper;
+
+    private String customPrefix;
+    private List<Pattern> pageExclusionPatterns;
+    private List<Pattern> assetExclusionPatterns;
+    private final List<String> excludedPages = new ArrayList<>();
```

Comment:
With this being `final` and thus only every added to in the code, I actually wonder if this consistently accrues pages across multiple requests.  This issue goes away if we remove this from being an instance var, as mentioned above.